zmq.hpp

```diff
@@ -30,6 +30,15 @@
 #ifndef NOMINMAX
 #define NOMINMAX
 #endif
+// To avoid the `(std::min)` workaround for when a min macro is defined
+#ifdef min
+#define WINDOWS_MIN min
+#undef min
+#endif
+#ifdef max
+#define WINDOWS_MAX max
+#undef max
+#endif
```

Comment:
This was initially to fix the formatting being different between clang-format 14 and 20 for the `(std::min)(x, y)` trick.

I ended up fixing it differently but this "guard" macros seems a bit cleaner than having to remember the `(std::min)` workaround. Let me know if I should keep it

(Still have to test if it works on Windows tho)